File: src/classes/Pin.ts
import { IPinConstructor } from '../interfaces/Pin.js';

export default class Pin {
	public standing: boolean;
	public number: number;

	constructor({ number }: IPinConstructor) {
		this.standing = true;
		this.number = number;
	}

	knockDown(): void {
		this.standing = false;
	}
}

File: Bowling/src/classes/Pin.cs
namespace Bowling;

public class Pin
{
  public bool IsStanding { get; set; }
  public int Number { get; set; }

  public Pin(int Number)
  {
    IsStanding = true;
    this.Number = Number;
  }

  public void KnockDown()
  {
    IsStanding = false;
  }
  public void Raise()
  {
    IsStanding = true;
  }
}
